src/elements/ExactSbend.H

```diff
@@ -196,33 +201,35 @@ namespace impactx::elements
             } else {
 
                // assign intermediate quantities
-               amrex::ParticleReal const pperp2 = powi<2>(pt)-2.0_prt * m_ibeta * pt - powi<2>(py)+1.0_prt;
-               amrex::ParticleReal const px2 = powi<2>(px);
+               T_Real const pperp2 = powi<2>(pt)-2.0_prt * m_ibeta * pt - powi<2>(py)+1.0_prt;
+               T_Real const px2 = powi<2>(px);
                // determine if particle lies within the domain of map definition
-               if (pperp2 <= px2)
+               auto const mask = pperp2 <= px2;
+               if (stdx::any_of(mask) == true)
                {
-                   amrex::ParticleIDWrapper{idcpu}.make_invalid();
+                   amrex::ParticleIDWrapper<T_IdCpu>{idcpu}.make_invalid(mask);
                } else
```

Comment:
Technically, we need to apply the else branch to all particles with `!mask`...

Reply:
if the `if (stdx::any_of(mask) == true)` needs to go so we still run the `else` for valid particles.